app/app.py

```diff
@@ -135,50 +143,56 @@
 
 
 if st.button("Generate Patches"):
-
-    # Set up git repo for git apply testing
-    try:
-        logging.debug("initializing repo in %s", repo_dir)
-        file_name = initialize_git_repository_and_store_files(
-            repo_dir, downstream_patch_text, downstream_before_text
-        )
-    except PermissionError as e:
-        logging.error("failed writing to %s", repo_dir, exc_info=True)
-        raise e
-
-    # Generate text
-    logging.info("requesting generation from model %s", model_id)
-    result = generate_text(
-        model_id, prompt, decoding_method, max_new_tokens, temperature, top_k, top_p
-    )
-
-    st.text(result)
-    # Parse text for patch
-    if extract_git_diff(result, repo_dir):
-        logging.debug("was able to extract the diff")
-        # Get file names in patch
-        patch_file_name = extract_path_from_text(result)
-        logging.debug("patch file name: %s", patch_file_name)
-        
-        # Get other metrics
-        test_results = output_quality_metrics(result, downstream_patch_text)
+    
+    success, diff = check_prompt_token_limit(model_id, prompt)
+    
+    if not success:
+        st.write(f"Prompt is {diff} tokens too long, please shorten it")
+    else:
+        # Set up git repo for git apply testing
+        try:
+            logging.debug("initializing repo in %s", repo_dir)
+            file_name = initialize_git_repository_and_store_files(
+                repo_dir, downstream_patch_text, downstream_before_text
+            )
+        except PermissionError as e:
+            logging.error("failed writing to %s", repo_dir, exc_info=True)
+            raise e
+
+        # Generate text
+        logging.info("requesting generation from model %s", model_id)
         
-        # Run tests
-        patch_tests = PatchValidationTests()
-        test_results.update(patch_tests.run_all_tests(file_name, patch_file_name, repo_dir))
-        logging.info("test results: %s", test_results)
-
-        logging.debug("deleting directory")
-        delete_directory(repo_dir)
-
-        # st.write(f"Total tests: {total_tests}")
-        st.write(f"Passes name tests: {test_results['name_test']}")
-        st.write(f"Passes git apply tests: {test_results['git_apply_test']}")
-        st.write(f"Passes compile test: {test_results['compile_test']}")
-        st.write(f"Levenshtein distance: {test_results['levenshtein_distance']}")
+        result = generate_text(
+            model_id, prompt, decoding_method, max_new_tokens, temperature, top_k, top_p
+        )
 
-    else:
-        logging.warning("failed to extract diff")
-        logging.debug("result given: %s", result)
-        st.write(f"diff not found")
+        st.text(result)
+        # Parse text for patch
+        if extract_git_diff(result, repo_dir):
```

Comment:
Similar to the above comment, if you instead check `if not extract_git_diff(result, repo_dir)` here then you can return and avoid indenting the rest of the function.

```suggestion
        if not extract_git_diff(result, repo_dir):
            logging.info("failed to extract diff")
            logging.debug("result given: %s", result)
            st.write(f"diff not found")
            return
       
        # rest of the function here can continue at the same indentation level
        logging.debug("was able to extract the diff")
          
        # Get file names in patch
        patch_file_name = extract_path_from_text(result)
        logging.debug("patch file name: %s", patch_file_name)

        # Get other metrics
        test_results = output_quality_metrics(result, downstream_patch_text)
```

Reply:
Complete